### ISSUE TITLE
Label Encoder dealing with Missing Labels

### ISSUE THREAD

**Issue by mherde** [scikit-activeml/scikit-activeml#6]
**Extending sklearn.preprocessing.LabelEncoder**

Sklearn based LabelEncoder cannot deal with missing labels and multi-dimensional label array. Extension includes:
- Parameter `classes`  defining all available class labels. If not provided, it is inferred during the fitting the encoder.
- Parameter `unlabeled_class`  representing a missing label. If not provided, `np.nan` is used as default value.
- Class labels are transformed to `0, ..., n_classes-1` and missing labels are encoded by `np.nan`.

Still missing:
- Function that checks whether the elements of `classes` together with `unlabeled_class` have the same data type.

**Comment by dakot** [comment]
@mherde Is pandas really necessary? we then would have an additional dependency

**Comment by mherde** [comment]
@dakot In a first stage pandas is not necessary. I removed it from the `requirements.txt` file.
